### PR TITLE
feat(extension): Add support for alternative Flashloan providers to migration extension

### PR DESCRIPTION
**Pull request by ckoopmann**

### CODE REVIEW

File: contracts/adapters/MigrationExtension.sol [hunk 1]
```diff
@@ -81,13 +83,15 @@ contract AaveMigrationExtension is BaseExtension, FlashLoanSimpleReceiverBase, I
     ITradeModule public immutable tradeModule;
     IDebtIssuanceModule public immutable issuanceModule;
     INonfungiblePositionManager public immutable nonfungiblePositionManager;
+    IMorpho public constant MORPHO = IMorpho(0xBBBBBbbBBb9cC5e90e3b3Af64bdAF62C37EEFFCb);
+    IBalancerVault public constant BALANCER = IBalancerVault(0xBA12222222228d8Ba445958a75a0704d566BF2C8);
```
Comment (pblivin0x) [inline contracts/adapters/MigrationExtension.sol:87]: for consistency, should we move other constructor inputs to this type of initialization?

File: contracts/adapters/MigrationExtension.sol [hunk 2]
```diff
@@ -293,6 +298,109 @@ contract AaveMigrationExtension is BaseExtension, FlashLoanSimpleReceiverBase, I
...
+    /**
+     * @dev Callback function for Balancer flashloan
+    */
+    function receiveFlashLoan(
+        IERC20[] memory tokens,
+        uint256[] memory amounts,
+        uint256[] memory feeAmounts,
+        bytes memory params
+    ) external {
+        require(msg.sender == address(BALANCER));
```
Comment (pblivin0x) [inline contracts/adapters/MigrationExtension.sol:347]: 👍

File: test/integration/ethereum/migrationExtensionBtc2x.spec.ts [hunk 3]
```diff
@@ -363,7 +363,7 @@ if (process.env.INTEGRATIONTEST) {
                     isUnderlyingToken0 = true;
 
                     // Subsidize 0.01 WBTC to the migration extension
-                    maxSubsidy = bitcoin(0.01);
+                    maxSubsidy = bitcoin(0.00);
```
Comment (pblivin0x) [inline test/integration/ethereum/migrationExtensionBtc2x.spec.ts:366]: 🔥

File: test/integration/ethereum/migrationExtensionEth2x.spec.ts [hunk 4]
```diff
@@ -342,7 +342,7 @@ if (process.env.INTEGRATIONTEST) {
                       aEthWeth.address,
                     );
                     const wrappedExchangeRate = preciseDiv(ether(1), wrappedPositionUnits);
-                    maxSubsidy = ether(3.205);
+                    maxSubsidy = ether(0.1);
```
Comment (pblivin0x) [inline test/integration/ethereum/migrationExtensionEth2x.spec.ts:345]: 🔥

**Review by pblivin0x — APPROVED** [review]
LGTM 💯

added some comments on optional cleanups

File: contracts/adapters/MigrationExtension.sol (hunk 1 above)
Comment (ckoopmann) [reply contracts/adapters/MigrationExtension.sol]: Actually went ahead and turned these two addresses into immutable values that can be specified in the constructor.

That way we can use it on l2 / other networks where the addresses might differ.

**Pull request merged**